crawl-ref/source/areas.cc

```diff
@@ -71,6 +79,18 @@ static bool _check_agrid_flag(const coord_def& p, areaprop f)
     return bool(_agrid(p) & f);
 }
 
+/// \brief Invalidate the area effect cache
+/// \details After calling this function, next check (of a coordinate for
+/// area effects) may trigger a re-calculation (of the caches by checking the
+/// player and all monsters for area effects). If \p recheck_new is true,
+/// next check will always trigger the re-calculation. Otherwise, next check
+/// will trigger the re-calculation only if the cache believes the level has
+/// existing area effects.
+/// \param recheck_new
+/// - If true, next check will always trigger a re-calculation. Use this if we
+///   have just created a new area effect.
+/// - If false, next check will trigger a re-calculation only if the level
+///   has existing area effects.
```

Comment:
This is a useful comment, though it could use a little work on phrasing and grammar. I'd probably go with something like:
```
// Invalidates the area effect cache, causing the next request for
// area effects to re-calculate which locations are covered by halos, etc.
// If recheck_new is false, the cache will only be invalidated if the level
// had existing area effects.
```

We can do the whole doxygen style comments if you want (though it's very unclear to me that anyone is ever generating or reading doxygen comments?). The key differences that I'm aiming for here are that we're explaining what's going in a shorter way, which is simpler for the reader to understand, and that we aren't telling the user what to do. It's rarely correct to say "use this if we've just created a new area effect" or things along that line. In most cases, we'd rather just explain what the options do and let the reader figure out how that applies to their use case.